### ISSUE TITLE
[py2py3] WMCore should pass all the unit tests run in py3 environment

### ISSUE THREAD

**Issue by mapellidario** [dmwm/WMCore#10422]
**Impact of the new feature**
Whole WMCore should be affected

**List**

* [x] PsetTweaks, #9739, #9740:
  - nothing needed
* [x] Utils, #9730, #9731
  - #10500 
* [ ] WMConfigCache, #9806, #9818
  - #10522
* [ ] WMCore/WMFactory - #9970, #9971
* [ ] WMQuality/CherrypyTestInit - #9972
* [ ] WMCore/Configuration - #9973, #9974
* [ ] WMCore/Credential - #10007, #10008
* [ ] WMCore/Algorithms - #10009, #10011
* [ ] WMCore/DataStructs - #10010, #10012
  - This is not fixed, but we have a specific issue to deal with this: #10169 
* [ ] slice 1 - #10055, #10057
* [ ] slice 2 - #10098, #10103
* [ ] slice 3 - #10114, #10117 
* [ ] slice 4 - #10115, #10165 
* [ ] slice 5 - #10116, #10262 
* [ ] slice 6 - #10127, #10277 
* [ ] slice 7 - #10131, #10284 
* [ ] slice 8 - #10132, #10289 
* [ ] slice 9 - #10133, #10290
* [ ] slice 10 - #10134, #10295
* [ ] slice 11 - #10135, #10299
* [ ] slice 12 - #10136, #10307
* [ ] slice 13 - #10137, #10310 
* [ ] slice 14 - #10138, #10313
* [ ] slice 15 - #10139, #10319
* [ ] slice 16 - #10140, #10321
* [ ] slice 17 - #10141, #10329
* [ ] slice 18 - #10142, #10331
* [ ] slice 19 - #10143, #10348
* [ ] slice 20 - #10144, #10349
* [ ] slice 21 - #10145, #10361
* [ ] slice 22 - #10146, #10362
* [ ] slice 23 - #10147, #10363 
* [ ] slice 24 - #10148, #10371 
* [ ] slice 25 - #10149, #10379 
* [ ] slice 26 - #10150, 

(list taken from #10440)

**Is your feature request related to a problem? Please describe.**
We testes the changes contained in the modernizazion slices ( [PRs with tag py2py3slice](https://github.com/dmwm/WMCore/pulls?q=label%3A%22py2py3+slice%22+is%3Apr) ) with the current jenkins setup, which has python2 only unit tests.

After we have a py3 environment, after we manage to integrate it with jenkins, we should check that all our changes pass all our unit tests run in python3.

**Describe the solution you'd like**

It is impossible to estimate at this stage how long this validation will take. After a quick private chat with Alan, we decided that 
- I create only this issue
- When we have the py3 env in jenkins, I look for the failures, fix some of those and open a PR. 
- Then, i fix some other features and open a new PR
- We go on like this until we do not have failing py3 unit tests. At that point, we close this issue.

It is likely that this will take us from a couple of weeks to one month, but since I already opened a ton of small issues, I think that having a single big issue is not going to spoil our tracking procedures.

**Describe alternatives you've considered**
It's not really wise to avoid this validation :grimacing: 

**Additional context**
This issue depends on #10227

**Comment by mapellidario** [comment]
Oh no, this should have been kept open! My bad, github auto-close-issue machinery tricked me
